osu.Game/Rulesets/Edit/Checks/CheckUnusedAudioAtEnd.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Rulesets.Edit.Checks.Components;
+using osu.Game.Rulesets.Objects;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckUnusedAudioAtEnd : ICheck
+    {
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Compose, "More than 20% unused audio at the end");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateUnusedAudioAtEnd(this),
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            double mappedLength = context.Beatmap.HitObjects.Last().GetEndTime();
```

Comment:
This is not necessarily correct in gamemodes with concurrent objects (i.e. mania).

We have a separate extension for this.

```suggestion
            double mappedLength = context.Beatmap.GetLastObjectTime();
```

Reply:
fixed